bindata/etcd/cluster-backup.sh

```diff
@@ -34,10 +34,14 @@ function backup_latest_kube_static_resources {
 
   LATEST_RESOURCE_DIRS=()
   for RESOURCE in "${RESOURCES[@]}"; do
-    # shellcheck disable=SC2012
-    LATEST_RESOURCE=$(ls -trd "${CONFIG_FILE_DIR}"/static-pod-resources/"${RESOURCE}"-[0-9]* | tail -1) || true
+    if [ ! -f /etc/kubernetes/manifests/"${RESOURCE}".yaml ]; then
+      echo "error finding manifests for ${RESOURCE}. please check if it is running."
+      exit 1
+    fi
+
+    LATEST_RESOURCE=$(jq -r '.spec.volumes | .[] | select (.name=="resource-dir") | .hostPath.path' /etc/kubernetes/manifests/"${RESOURCE}".yaml) || true
```

Comment:
just something to keep in mind when we backport only recently 4.7? is our static pod JSON on disk. I believe the change was introduced through library-go bump, have not dug very deep. But then this will fail in say 4.4 where it is YAML.

Reply:
Thanks. Changed it to a simple `grep` command so that it works in all releases regardless of it the manifest files are in json or yaml format.